### PR TITLE
Item code generation

### PR DESCRIPTION
**Pull request by Trompettesib**

### CODE REVIEW

**Review by huntears — APPROVED** [review]

File: cubic-server/protocol_id_converter/itemConverter.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    });
+    if (item == this->_items.end()) {
+        LERROR("Item not found in palette (name: " << name << ")");
+        return 0;
+    }
+    return item->protocolId;
+}
+
+std::string Items::ItemConverter::fromProtocolIdToItem(ItemId id) const {
+    auto item = std::find_if(this->_items.begin(), this->_items.end(), [&id](Items::InternalItem i) {
```
Comment (STMiki) [inline cubic-server/protocol_id_converter/itemConverter.cpp:33]:
```suggestion
    auto item = std::find_if(this->_items.begin(), this->_items.end(), [&id](const Items::InternalItem &i) {
```

File: cubic-server/protocol_id_converter/itemConverter.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    for(auto item : file["minecraft:item"]["entries"].items()) {
+        Items::InternalItem i;
+        i.name = item.key();
+        i.protocolId = item.value()["protocol_id"];
+        this->_items.push_back(i);
+    }
+}
+
+Items::ItemId Items::ItemConverter::fromItemToProtocolId(std::string name) const {
+    auto item = std::find_if(this->_items.begin(), this->_items.end(), [&name](Items::InternalItem i) {
```
Comment (STMiki) [inline cubic-server/protocol_id_converter/itemConverter.cpp:22]:
```suggestion
    auto item = std::find_if(this->_items.begin(), this->_items.end(), [&name](const Items::InternalItem &i) {
```

File: cubic-server/protocol_id_converter/itemConverter.cpp [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    nlohmann::json file = nlohmann::json::parse(std::ifstream(path));
+    for(auto item : file["minecraft:item"]["entries"].items()) {
+        Items::InternalItem i;
+        i.name = item.key();
+        i.protocolId = item.value()["protocol_id"];
+        this->_items.push_back(i);
+    }
+}
+
+Items::ItemId Items::ItemConverter::fromItemToProtocolId(std::string name) const {
```
Comment (STMiki) [inline cubic-server/protocol_id_converter/itemConverter.cpp:21]:
```suggestion
Items::ItemId Items::ItemConverter::fromItemToProtocolId(const std::string &name) const {
```

**Review by STMiki — CHANGES_REQUESTED** [review]
Reference for the win ! 💪

File: cubic-server/types.hpp [hunk 4]
```diff
@@ -14,6 +14,7 @@ typedef int64_t Seed;
 
 typedef int32_t GlobalBlockId;
```
Comment (STMiki) [inline cubic-server/types.hpp:15]: Honnestly, we could just remove the `Global`, will add that to #117

**Review by STMiki — DISMISSED** [review]
:ok_man:

**Review by huntears — APPROVED** [review]

File: cubic-server/protocol_id_converter/blockStates.cpp [hunk 5]
```diff
@@ -34,7 +34,7 @@ void Blocks::GlobalPalette::initialize(std::string path) {
     }
 }
 
-GlobalBlockId Blocks::GlobalPalette::fromBlockToProtocolId(Blocks::Block block) const {
+BlockId Blocks::GlobalPalette::fromBlockToProtocolId(Blocks::Block block) const {
```
Comment (STMiki) [inline cubic-server/protocol_id_converter/blockStates.cpp:37]: :upside_down_face: 
```suggestion
BlockId Blocks::GlobalPalette::fromBlockToProtocolId(const Blocks::Block &block) const {
```

File: cubic-server/protocol_id_converter/blockStates.hpp [hunk 6]
```diff
@@ -46,13 +46,13 @@ namespace Blocks {
              * @param block The block to convert
              * @return The protocol id of the block
              */
-            GlobalBlockId fromBlockToProtocolId(const Block block) const;
+            BlockId fromBlockToProtocolId(const Block block) const;
```
[the remark this answers is not in the document]
Comment (STMiki) [reply cubic-server/protocol_id_converter/blockStates.hpp:49]:
```cpp
            void initialize(const std::string &path = "blocks.json");
```

**Review by huntears — APPROVED** [review]

**Review by STMiki — CHANGES_REQUESTED** [review]
Missed theses :grinning:

**Review by STMiki — APPROVED** [review]
:100:

**Review by huntears — APPROVED** [review]

**Pull request merged**